packages/airnode-node/src/evm/handlers/process-transactions.test.ts

```diff
@@ -18,114 +19,184 @@ mockEthers({
+  test.each(['1', '2'])(
+    'fetches the gas price, assigns nonces and submits transactions - txType: %d',
+    async (txType) => {
+      const initialConfig = fixtures.buildConfig();
+      const chainOptions = { txType } as ChainOptions;
+      const chains = initialConfig.chains.map((chain) => ({
+        ...chain,
+        chainOptions,
+      }));
+      const config = {
+        ...initialConfig,
+        chains,
+      } as Config;
+      jest.spyOn(fs, 'readFileSync').mockReturnValue(JSON.stringify(config));
+
+      const gasPriceSpy = jest.spyOn(ethers.providers.JsonRpcProvider.prototype, 'getGasPrice');
+      const blockSpy = jest.spyOn(ethers.providers.JsonRpcProvider.prototype, 'getBlock');
+      const gasTarget = (() => {
+        if (txType === '1') {
+          const gasPrice = ethers.BigNumber.from(1000);
+          gasPriceSpy.mockResolvedValue(gasPrice);
+          return { gasPrice };
+        }
+
+        const baseFeePerGas = ethers.BigNumber.from(1000);
+        blockSpy.mockResolvedValue({ baseFeePerGas } as ethers.providers.Block);
+        const maxPriorityFeePerGas = BigNumber.from(PRIORITY_FEE);
+        const maxFeePerGas = baseFeePerGas.mul(BASE_FEE_MULTIPLIER).add(maxPriorityFeePerGas);
+
+        return { maxPriorityFeePerGas, maxFeePerGas } as GasTarget;
+      })();
+
+      const balanceSpy = jest.spyOn(ethers.providers.JsonRpcProvider.prototype, 'getBalance');
+      const balance = ethers.BigNumber.from('5000000000000000000');
+      balanceSpy.mockResolvedValueOnce(ethers.BigNumber.from(balance));
+
+      estimateGasWithdrawalMock.mockResolvedValueOnce(ethers.BigNumber.from(50_000));
+      fulfillWithdrawalMock.mockResolvedValueOnce({
+        hash: '0xcbb3f9dc6a24e8b6f5427dcf960b1da01c3df0636cb25a292f8dcaad78755c8d',
+      });
+      staticFulfillMock.mockResolvedValueOnce({ callSuccess: true });
+      fulfillMock.mockResolvedValueOnce({
+        hash: '0xad33fe94de7294c6ab461325828276185dff6fed92c54b15ac039c6160d2bac3',
+      });
+
+      const apiCall = fixtures.requests.buildSubmittableApiCall({
+        sponsorAddress: '0x69e2B095fbAc6C3f9E528Ef21882b86BF1595181',
+      });
+      const withdrawal = fixtures.requests.buildWithdrawal({
+        sponsorAddress: '0x99bd3a5A045066F1CEf37A0A952DFa87Af9D898E',
+      });
+      const requests: GroupedRequests = {
+        apiCalls: [apiCall],
+        withdrawals: [withdrawal],
+      };
+      const transactionCountsBySponsorAddress = {
+        '0x69e2B095fbAc6C3f9E528Ef21882b86BF1595181': 79,
+        '0x99bd3a5A045066F1CEf37A0A952DFa87Af9D898E': 212,
+      };
+      const initialState = fixtures.buildEVMProviderState({
+        requests,
+        transactionCountsBySponsorAddress,
+      });
+
+      const state = {
+        ...initialState,
+        config,
+        settings: {
+          ...initialState.settings,
+          chainOptions,
+        },
+      } as ProviderState<EVMProviderState>;
```

Comment:
This makes it unhappy (it becomes a problem in the call to `processTransactions`.

Reply:
It is unhappy because `txType` is string and not `'1' | '2'`.
If you do:
- test.each(['1', '2'] as const)... (add as const)
- async (txType) => ... (remove :string)
- do what is in the suggestion

it should work.

Reply:
Thank you :), I've learnt something, obvious now :)